src/Function.cpp

```diff
@@ -248,6 +338,103 @@ Function::Function(const std::string &n) : contents(new FunctionContents) {
     contents.ptr->name = n;
 }
 
+void deep_copy_update_definition_helper(IntrusivePtr<FunctionContents> &contents,
+                                        UpdateDefinition &dst,
+                                        const UpdateDefinition &src,
+                                        DeepCopyMap &copied_map) {
+    dst.values = src.values;
+    dst.args = src.args;
+    dst.schedule = Schedule();
+    src.schedule.deep_copy(dst.schedule, copied_map);
```

Comment:
Should this be a constructor? i.e. this would look like:

    dst.schedule = Schedule(src.schedule, copied_map);

It would reduce ambiguity in whether deep_copy copies from or to, and reduces the risk of leaving behind some fields from the old schedule.